File: NoteBook/src/main/java/SkuratovichSiarhey/by/ComandPattern/Comands/ShowNotesCommand.java
package SkuratovichSiarhey.by.ComandPattern.Comands;

import SkuratovichSiarhey.by.ComandPattern.Command;
import SkuratovichSiarhey.by.NoteBook;
import SkuratovichSiarhey.by.NoteBookConsoleView;

/**
 * Implements users command of showing full list of notes.
 */
public class ShowNotesCommand implements Command {
  private NoteBook noteBook;

  public ShowNotesCommand(NoteBook noteBook) {
    this.noteBook = noteBook;
  }

  /**
   * Shows full list of notes.
   */
  public void execute() {
    NoteBookConsoleView noteBookConsoleView = new NoteBookConsoleView(noteBook);
    noteBookConsoleView.print("full");
  }
}

File: TaskAboutNoteBook/ShowNotesCommand.cs
using System;


namespace TaskAboutNoteBook
{
  /// <summary>
  /// Shows notes in notebook.
  /// </summary>
  public class ShowNotesCommand : Command
  {
    private NoteBookConsoleView noteBookConsoleView;

    /// <summary>
    /// Constructor.
    /// </summary>
    /// <param name="noteBookConsoleView">Instance of NoteBookConsoleView.</param>
    public ShowNotesCommand(NoteBookConsoleView noteBookConsoleView)
    {
      this.noteBookConsoleView = noteBookConsoleView;
;     
    }

    /// <summary>
    /// Executes command.
    /// </summary>
    public void Execute()
    {
      Console.WriteLine(noteBookConsoleView.Print());
    }
  }
}
